src/core/timer.c

```diff
@@ -893,6 +906,64 @@ static int timer_can_start(Unit *u) {
         return 1;
 }
 
+static void activation_event_info_timer_serialize(ActivationEventInfo *info, FILE *f) {
+        ActivationEventInfoTimer *t = ACTIVATION_EVENT_INFO_TIMER(info);
+
+        assert(info);
+        assert(f);
+        assert(t);
+
+        (void) serialize_dual_timestamp(f, "activation-event-info-timer-last-trigger", &t->last_trigger);
+}
+
+static int activation_event_info_timer_deserialize(const char *key, const char *value, ActivationEventInfo **info) {
+        int r;
+
+        assert(key);
+        assert(value);
+
+        if (streq(key, "activation-event-info-timer-last-trigger")) {
+                if (!info || !*info)
+                        return -EINVAL;
+
+                ActivationEventInfoTimer *t = ACTIVATION_EVENT_INFO_TIMER(*info);
+                if (!t)
+                        return -EINVAL;
+
+                r = deserialize_dual_timestamp(value, &t->last_trigger);
+                if (r < 0)
+                        return r;
+
+                return 0;
+        }
+
+        return -EINVAL;
+}
+
+static int activation_event_info_timer_format(ActivationEventInfo *info, char ***strv) {
+        ActivationEventInfoTimer *t = ACTIVATION_EVENT_INFO_TIMER(info);
+        char *s;
+
+        assert(info);
+        assert(strv);
+        assert(t);
+
+        if (!dual_timestamp_is_set(&t->last_trigger))
+                return 0;
+
+        s = strjoin("TRIGGER_TIMER_REALTIME=", FORMAT_TIMESTAMP(t->last_trigger.realtime));
```

Comment:
hmm, this would format the timestamps in our own format. I figure it might be nice to format it as integer simply, i.e. USEC_FMT. Should be simpler for people to consume I think.

Maybe rename TRIGGER_TIMER_REALTIME_USEC= also? i.e. make the unit clear, once it is just a formatted integer.